src/slm/slmbuild/slmbuild.cpp

```diff
@@ -233,13 +233,20 @@ getParameters(int argc, char* argv[])
 int
 main(int argc, char* argv[])
 {
+    FILE* fp = NULL;
+
     getParameters(argc, argv);
 
     TSIMWordId * ngram = new TSIMWordId[N + 1];
     CSlmBuilder::FREQ_TYPE freq;
 
     printf("Reading and Processing raw idngram..."); fflush(stdout);
-    FILE *fp = fopen(inputfilename, "rb");
+
+    if ((fp = fopen(inputfilename, "rb")) == NULL) {
+        printf("Can not open raw idngram file %s\n", inputfilename);
+        return 99;
```

Comment:
is `99` a document retval? if not, instead of using a random retval, might want to use `EXIT_FAILURE`.

Reply:
It is used in another slm tool and that's where this code snippet comes.